lib/runner.js

```diff
@@ -303,9 +303,8 @@ class Runner extends Emittery {
 
 	async runTest(task, contextRef) {
 		let hooksAndTestOk = false;
-
 		const hookSuffix = ` for ${task.title}`;
-		if (await this.runHooks(this.tasks.beforeEach, contextRef, hookSuffix)) {
+		if (await this.runHooks(this.tasks.beforeEach, contextRef, task.args, hookSuffix)) {
```

Comment:
I think we should pass the `task.implementation` as well. That way, the hook can identify the macro that's being used.